src/move/econia/sources/user.move

```diff
@@ -409,6 +409,78 @@ module econia::user {
         size: u64
     }
 
+    /// Table keys are market account IDs.
+    struct MarketEventHandles has key {
+        cancel_posted_order_events:
+            Tablist<u128, EventHandle<CancelPostedOrderEvent>>,
+        change_order_size_events:
+            Tablist<u128, EventHandle<ChangeOrderSizeEvent>>,
+        fill_events: Tablist<u128, EventHandle<FillEvent>>,
+        place_limit_order_events:
+            Tablist<u128, EventHandle<PlaceLimitOrderEvent>>,
+        place_market_order_events:
+            Tablist<u128, EventHandle<PlaceMarketOrderEvent>>
+    }
+
+    struct CancelPostedOrderEvent has copy, drop, store {
```

Comment:
As per @qdrs, I don't think `Posted` here adds anything

Reply:
`Posted` to make it explicit that this event is emitted only when a posted order is cancelled, representing a mutation to order book state.

This is in contrast, for example, to

* A swap that eats through all price levels before fully filling, such that the matching engine halts.
* A limit order that fills multiple times across the spread and then self-matches with the `CANCEL_TAKER` flag, such that the matching engine halts and then the limit order does not post. 

Here the result of each order is fully captured by the combination of `FillEvent`s and, respectively, the `PlaceSwapOrderEvent` or `PlaceLimitOrderEvent` (which includes a `size_posted` field).

Since the logic would be more complex and the events would be redundant if we were to also to emit cancel events for these kinds of cases, I chose to clarify that only posted order cancellations emit events.

Your thoughts?

Reply:
Then how does the user who posted that limit order know that the order stopped filling because it reached another one of their orders?

The user needs to be able to determine the current status of their orders based on the events they receive through these event handles. The combination of these events do not offer enough information to definitively determine order status.

Reply:
> Then how does the user who posted that limit order know that the order stopped filling because it reached another one of their orders?

Also, just to clarify, I use the term "post" to denote an order that is on the book waiting to be filled.

I am confused by your above statement, as I would expect the word "submit" where you used the word "post": if an order stops filling due to a self-match I wouldn't expect it to post

https://www.independentreserve.com/blog/knowledge-base/what-is-a-post-only-order-type

Reply:
Yes, I should have used "placed" instead of "posted" there. Apologies for the imprecise language.

> Presently, the user knows how much posted to the order book via the size_posted field of PlaceLimitOrderEvent, which is not necessarily the same as the size field.

That's not the case I'm asking about, I'm asking about limit orders that never make it to the prior book because they encounter a self match prior to doing so.

I think the partially maker, partially taker case is fine now. Although I do maintain that we should just have `remaining_size` / `amount_filled` separate from `size` like every other exchange on the planet, instead of the `size_posted` field here.

Reply:
> I believe you may be looking for something like the following:
> 
> - Cancel reasons CANCEL_REASON_SELF_MATCH_TAKER and CANCEL_REASON_SELF_MATCH_MAKER corresponding to cancellation of the taker side and/or the maker side depending on self-match behavior
> - Cancel reason CANCEL_REASON_BOOK_EMPTY for when the matching engine halts due to no more possible fills
> - Cancel reason CANCEL_REASON_IMMEDIATE_OR_CANCEL for an IOC order that does not fill across the spread

Where would events with these values be emitted? Under the current event schema, cancel events are only emitted for maker events. This doesn't provide users with enough information to determine the status of their orders via events, as I noted above.

Reply:
> I think the partially maker, partially taker case is fine now. Although I do maintain that we should just have `remaining_size` / `amount_filled` separate from `size` like every other exchange on the planet, instead of the `size_posted` field here.

@bogberry `PlaceLimitOrderEvent.size_posted` changed to `remaining_size` per https://github.com/econia-labs/econia/pull/321#issuecomment-1635034144

> Where would events with these values be emitted? Under the current event schema, cancel events are only emitted for maker events. This doesn't provide users with enough information to determine the status of their orders via events, as I noted above.

Also per https://github.com/econia-labs/econia/pull/321#issuecomment-1635034144 , `CancelPostedOrderEvent` updated to `CancelOrderEvent` with relevant cancellation reason emission.